crates/gen/src/types/ntstatus.rs

```diff
@@ -0,0 +1,34 @@
+use super::*;
+
+pub fn gen_ntstatus() -> TokenStream {
+    quote! {
+        #[repr(transparent)]
+        #[derive(::std::default::Default, ::std::clone::Clone, ::std::marker::Copy, ::std::cmp::PartialEq, ::std::cmp::Eq, ::std::fmt::Debug)]
+        pub struct NTSTATUS(pub u32);
+
+        impl NTSTATUS {
+            #[inline]
+            pub const fn is_ok(self) -> bool {
+                self.0 == 0
```

Comment:
This deviates from the `NT_SUCCESS` macro, which essentially does `self.0 >= 0`. I'm worried that keeping it like this will damage error propagation ergonomics through the `?` operator.

Reply:
Yes, good point.